src/olympia/shelves/forms.py

```diff
@@ -0,0 +1,51 @@
+import requests
+
+from rest_framework.reverse import reverse as drf_reverse
+
+from django import forms
+from django.core.exceptions import ValidationError
+from django.utils.translation import ugettext_lazy as _
+
+from olympia.shelves.models import Shelf
+
+
+class ShelfForm(forms.ModelForm):
+    class Meta:
+        model = Shelf
+        fields = '__all__'
```

Comment:
define fields in `ShelfAdmin` instead and explicitly list them rather than `__all__` - we probably don't want to see created or modified.